src/screens/Profiles/MediaPages/pages/Summary/component.js

```diff
@@ -41,42 +43,64 @@ class SummaryComponent extends PureComponent {
   }
 
   state = {
-    loading: true,
+    feed: [],
+    loadingFeed: true,
+    loadingNextFeed: false,
   }
 
   componentDidMount() {
-    this.fetchFeed();
+    this.fetchFeed({ reset: true });
   }
 
   formatData(data, numberOfItems = 12) {
     if (!data) return [];
     return data.sort((a, b) => a.number - b.number).slice(0, numberOfItems);
   }
 
-  fetchFeed = async () => {
+  canFetchNextFeed = true;
+  feedCursor = undefined;
+  fetchFeed = async ({ reset = false } = {}) => {
     const { type, id } = this.props.media;
     const endpoint = type.charAt(0).toUpperCase() + type.slice(1);
 
-    this.setState({ loading: true });
+    if (reset) {
+      this.canFetchNextFeed = true;
+      this.feedCursor = undefined;
+      this.setState({ loadingFeed: true });
+    } else if (this.canFetchNextFeed) {
+      this.setState({ loadingNextFeed: true });
+    }
 
+    let data = [];
     try {
       const result = await Kitsu.one('mediaFeed', `${endpoint}-${id}`).get({
         include: 'media,actor,unit,subject,target,target.user,target.target_user,target.spoiled_unit,target.media,target.target_group,subject.user,subject.target_user,subject.spoiled_unit,subject.media,subject.target_group,subject.followed,subject.library_entry,subject.anime,subject.manga,subject.uploads,target.uploads',
         filter: {
           kind: 'posts',
         },
         page: {
+          cursor: this.feedCursor,
           limit: 10,
         },
       });
 
+      // I need to read the cursor value out of the 'next' link in the result.
+      this.canFetchNextFeed = !isEmpty(result && result.links && result.links.next);
+      const url = new URL(result.links.next, true);
+      this.feedCursor = url.query['page[cursor]'];
+
+      // Filter out non-post activities
       const feed = preprocessFeed(result).filter(i => i.type === 'posts');
-      this.setState({
-        feed,
-        loading: false,
-      });
+      data = reset ? [...feed] : [...this.state.feed, ...feed];
```

Comment:
You might want to use `uniqBy` on `id` in the feed data when you're setting it.
```
const feed = preprocessFeed(result).filter(i => i.type === 'posts');
data = reset ? [...feed] : [...this.state.feed, ...feed];
data = uniqBy(data, 'id');
```

This is just so we don't get a duplicate feed post which can cause a crash on a device.